app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavouriteButton.kt

```diff
@@ -0,0 +1,42 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun FavouriteButton(
+    color: Color,
+    modifier: Modifier = Modifier
+) {
+    var isFavourite by remember { mutableStateOf(false) }
```

Comment:
isFavorite should be an input 
With the current design the caller of this function cannot change the state of the heart icon programmatically. 
In general, it is a good practice to pass the state into the composable functions to make the UI as dumb as possible. This goes hand in hand with unidirectional data flow which the devs had in mind when Compose was being designed.
You can read more [here](https://developer.android.com/jetpack/compose/mental-model)